### PR TITLE
Take a single bit in expression to select charging port status.

### PR DESCRIPTION
**Pull request by abstractionnl**
Take a single bit in expression to select charging port status. 
Add vehicle profiles for Kia Niro and more properties to Hyundai Ioniq

Expressions now support using B14:5 to take the 5th bit (32) from the 14 byte in the payload.

Would be nice if these could show up as a binary sensors in home assistant.

### CODE REVIEW

**Comment by meatpiHQ** [comment]
I assume that you have tested the niro-phev.json ?

**Comment by abstractionnl** [comment]
Yes, I own both cars and verified the pids and expressions.

For Kia Niro PHEV, when the cable is connected, B14 changes from 0 to 128, when charging to 131. Car on contact or driving seems to be 3 (BMS active?). I don't think there is a real status bit for charging.
Hyundai Ioniq EV has cable connected at 32 and charging at 163 (bit 5: ac charger connected, bit 6: dc charger connected, bit 7 charging).

Do you think we should add more parameters?

**Review by meatpiHQ — COMMENTED** [review]
any reason you changed `"pid": "21057"` to `"pid": "2105"` ?

The 7 at the end tells it the number of expected response frames.

File: vehicle_profiles/hyundai/ioniq-2017.json [hunk 1]
```diff
@@ -3,13 +3,33 @@
   "init": "ATSP6;ATSH7E4;ATST96;",
   "pids": [
     {
-      "pid": "21057",
+      "pid": "2105",
       "parameters": [
         {
-          "name": "SOC_DISPLAY",
+          "name": "soc_display",
```
Comment (jay-oswald) [inline vehicle_profiles/hyundai/ioniq-2017.json:9]: Changing the name here will break things for people already using it, best to maintain the legacy name

Comment (happyjaxx) [reply vehicle_profiles/hyundai/ioniq-2017.json:9]: agreed, I more of a camelCase type but I'm more of a stability kinda guy... I'm already ingesting data using the capitalized entry

Comment (abstractionnl) [reply vehicle_profiles/hyundai/ioniq-2017.json]: Too bad there is no real standard for this. Maybe we should also add a friendly name too, to use in Home Assistant.
I reverted this backward incompatible change.

**Review by happyjaxx — APPROVED** [review]
would be happy to give this a shot

**Comment by jay-oswald** [comment]
@abstractionnl  see validation errors from the workflow

**Comment by abstractionnl** [comment]
> @abstractionnl see validation errors from the workflow

@jay-oswald I added class and unit fields, but it feels a bit redundant having to add these properties even tough they are empty.
I was thinking using class "plug" and "battery_charging" for them, but that would require them to be binary sensors, something I'd like to do in a next PR.

**Pull request merged**